cpp/src/io/comp/debrotli.cu

```diff
@@ -269,51 +281,50 @@ static __device__ uint32_t getvlc(debrotli_state_s *s, const uint16_t *lut)
   return vlc;
 }
 
+static auto __device__ allocation_size(uint32_t bytes) { return (bytes + 3) & ~3; }
+
 /// Alloc bytes from the local (shared mem) heap
-static __device__ uint8_t *local_alloc(debrotli_state_s *s, uint32_t bytes)
+static __device__ void *local_alloc(debrotli_state_s *s, uint32_t bytes)
 {
-  int heap_used = s->heap_used;
-  int len       = (bytes + 7) >> 3;
+  int heap_used  = s->heap_used;
+  auto const len = allocation_size(bytes);
   if (heap_used + len <= s->heap_limit) {
-    uint8_t *ptr = reinterpret_cast<uint8_t *>(&s->heap[heap_used]);
     s->heap_used = (uint16_t)(heap_used + len);
-    return ptr;
+    return &s->heap[heap_used];
   } else {
     return nullptr;
   }
 }
 
 /// Shrink the size of the local heap, returns ptr to end (used for stack-like intermediate
 /// allocations at the end of the heap)
-static __device__ uint8_t *local_heap_shrink(debrotli_state_s *s, uint32_t bytes)
+static __device__ void *local_heap_shrink(debrotli_state_s *s, uint32_t bytes)
 {
   int heap_used  = s->heap_used;
   int heap_limit = s->heap_limit;
-  int len        = (bytes + 7) >> 3;
+  auto const len = allocation_size(bytes);
   if (heap_limit - len >= heap_used) {
     heap_limit -= len;
     s->heap_limit = (uint16_t)heap_limit;
-    return reinterpret_cast<uint8_t *>(&s->heap[heap_limit]);
+    return &s->heap[heap_limit];
   } else {
     return nullptr;
   }
 }
 
 static __device__ void local_heap_grow(debrotli_state_s *s, uint32_t bytes)
 {
-  int len        = (bytes + 7) >> 3;
+  auto const len = allocation_size(bytes);
   int heap_limit = s->heap_limit + len;
   s->heap_limit  = (uint16_t)heap_limit;
 }
 
 /// Alloc memory from the fixed-size heap shared between all blocks (thread0-only)
-static __device__ uint8_t *ext_heap_alloc(uint32_t bytes,
-                                          uint8_t *ext_heap_base,
-                                          uint32_t ext_heap_size)
+static __device__ void *ext_heap_alloc(uint32_t bytes, void *ext_heap_base, uint32_t ext_heap_size)
 {
-  uint32_t len                = (bytes + 0xf) & ~0xf;
-  volatile uint32_t *heap_ptr = reinterpret_cast<volatile uint32_t *>(ext_heap_base);
-  uint32_t first_free_block   = ~0;
+  uint32_t len              = (bytes + 0xf) & ~0xf;
+  auto heap_ptr             = static_cast<volatile uint32_t *>(ext_heap_base);
+  uint32_t first_free_block = ~0;
   for (;;) {
```

Comment:
```suggestion
  while (true) {
```

Reply:
`for (;;)` requires the reader know the ins-and-outs of a for-loop, where as `while (true)` is obvious, even to new devs.